src/main/java/org/dataone/hashstore/hashfs/HashUtil.java

```diff
@@ -44,27 +47,31 @@ public boolean validateAlgorithm(String algorithm) {
      * 
      * @param prefix
      * @param directory
-     * @return
+     * 
+     * @return Temporary file (File) ready to write into
      * @throws IOException
+     * @throws SecurityException
      */
-    public File generateTmpFile(String prefix, File directory) throws IOException {
+    public File generateTmpFile(String prefix, File directory) throws IOException, SecurityException {
         String newPrefix = prefix + "-" + System.currentTimeMillis();
```

Comment:
If two threads create temp files at the same time, they may get the same file name. So this causes collision.

Reply:
Thank you - I added an additional random number:
```
Random random = new Random();
int randomNumber = random.nextInt(900000);
String newPrefix = prefix + "-" + System.currentTimeMillis() + randomNumber;
```